### PR TITLE
Add pybind for T PointCloud and fix associated bugs

### PR DESCRIPTION
**Pull request by errissa**
Prior to this PR the following would crash when trying to visualize T Point Cloud (last line):

```
import open3d as o3d

lsrc = o3d.io.read_point_cloud("/home/renes/Downloads/lounge_vh.ply")
tsrc = o3d.t.io.read_point_cloud("/home/renes/Downloads/lounge_vh.ply")

o3d.visualization.draw([lsrc])
o3d.visualization.draw([tsrc])

```

---
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3345)

### CODE REVIEW

**Review by prewettg — APPROVED** [review]
> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3345)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @reyanshsolis)

File: cpp/open3d/visualization/visualizer/O3DVisualizer.cpp [hunk 1]
```diff
@@ -930,7 +933,17 @@ struct O3DVisualizer::Impl {
...
-        scene->AddGeometry(name, geom.get(), mat);
+        // Do we have a tgeometry or a geometry?
+        if (geom) {
+            scene->AddGeometry(name, geom.get(), mat);
+        } else if (tgeom && valid_tpcd) {
+            scene->AddGeometry(name, valid_tpcd, mat);
+        } else {
+            utility::LogWarning(
+                    "No valid geometry specified to O3DViewer. Only supported "
+                    "geometries are Geometry3D and TGeometry PointClouds");
```
Comment (reyanshsolis) [inline cpp/open3d/visualization/visualizer/O3DVisualizer.cpp:944]: We may add a `.` after `PointClouds`.

File: cpp/open3d/visualization/visualizer/O3DVisualizer.cpp [hunk 2]
```diff
@@ -829,6 +829,8 @@ struct O3DVisualizer::Impl {
         bool is_default_color;
         bool no_shadows = false;
         Material mat;
+        t::geometry::PointCloud* valid_tpcd = nullptr;
```
Comment (reyanshsolis) [inline cpp/open3d/visualization/visualizer/O3DVisualizer.cpp:832]: Is it safe to use a pointer to `t::geometry::PointCloud`?
In the future may we change the pointer-based interface to a reference-based ?

**Review by errissa — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3345)* status: 2 of 3 files reviewed, 2 unresolved discussions (waiting on @prewettg and @reyanshsolis)

___
*[cpp/open3d/visualization/visualizer/O3DVisualizer.cpp, line 832 at r1](https://reviewable.io/reviews/intel-isl/open3d/3345#-MZJ497UmjvPzi3k3JY1-r1-832:-MZKd1emCrYDzUlhEAwT:b1pv9fn) ([raw file](https://github.com/intel-isl/open3d/blob/0df3a02bd8baba78e457d8bafcc8271ba86f1f40/cpp/open3d/visualization/visualizer/O3DVisualizer.cpp#L832)):*
<details><summary><i>Previously, reyanshsolis (Rishabh Singh) wrote…</i></summary><blockquote>

Is it safe to use a pointer to `t::geometry::PointCloud`?
In the future may we change the pointer-based interface to a reference-based ?
</blockquote></details>

It is safe to use a pointer here. The lower layers copy any data that may not survive when the `shared_ptr` protecting the pointer goes out of scope. It would be hard to use a reference-based interface here because the interface takes a base class.
___
*[cpp/open3d/visualization/visualizer/O3DVisualizer.cpp, line 944 at r1](https://reviewable.io/reviews/intel-isl/open3d/3345#-MZJ497UmjvPzi3k3JY1-r1-944:-MZKd0dkEcdMcW0iZZxd:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/0df3a02bd8baba78e457d8bafcc8271ba86f1f40/cpp/open3d/visualization/visualizer/O3DVisualizer.cpp#L944)):*
<details><summary><i>Previously, reyanshsolis (Rishabh Singh) wrote…</i></summary><blockquote>

We may add a `.` after `PointClouds`.
</blockquote></details>

Done.

**Review by reyanshsolis — APPROVED** [review]

**Pull request merged**